scipy/spatial/_lloyd_algorithm.py

```diff
@@ -0,0 +1,234 @@
+"""Lloyd-Max algorithm."""
+from __future__ import annotations
+
+import numpy as np
+from typing import (
+    Optional,
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import (
+        DecimalNumber, IntNumber
+    )
+
+from scipy.spatial import distance, Voronoi
+
+
+__all__ = ['lloyd_centroidal_voronoi_tessellation']
+
+
+def _l1_norm(points: np.ndarray) -> float:
+    l1 = distance.cdist(points, points, 'cityblock')
+    return np.min(l1[l1.nonzero()])
```

Comment:
similar comment here, since I'm poking around in the PR a bit today--I think we can nicely simplify this to avoid calculating i.e., an extra diagonal only to discard it:

```diff
--- a/scipy/spatial/_lloyd_algorithm.py
+++ b/scipy/spatial/_lloyd_algorithm.py
@@ -20,8 +20,7 @@ __all__ = ['lloyd_centroidal_voronoi_tessellation']
 
 
 def _l1_norm(points: np.ndarray) -> float:
-    l1 = distance.cdist(points, points, 'cityblock')
-    return np.min(l1[l1.nonzero()])
+    return distance.pdist(points, 'cityblock').min()
```